src/scripts/components/widgets/modals/Modal.js

```diff
@@ -65,23 +75,63 @@ const SVG = styled.svg`
 
 const Content = styled.div`
   height: 100%;
-  padding: 40px 46px;
+  padding: 44px 48px;
   width: 100%;
 `
 
+export const ModalContentWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  height: 100%;
+  width: 100%;
+`
+
+export const ModalScrollContent = styled.div`
+  flex: 1 1 100%;
+
+  @media (max-width: 767px) {
+    overflow-x: hidden;
+    overflow-y: scroll;
+  }
+`
+
 class Modal extends Component<Props, void> {
+  renderModal () {
+    const { modalContent } = this.props
+    switch (modalContent) {
+      case 'ModalStake':
+        return <ModalStake />
+    }
+  }
+
+  handleKeydown (event) {
+    if (event.keyCode === 27) {
+      this.props.closeModal()
+    }
+  }
+
+  componentDidMount () {
+    document.addEventListener('keydown', this.onKeydown)
+  }
+
+  componentDidUnmount () {
```

Comment:
Sorry I had a typo in my example, this is `componentWillUmnount`. There is no `componentDidUnmount`. My bad!